MaterialDesignThemes.Wpf/DrawerHost.cs

```diff
@@ -406,6 +409,112 @@ public bool BottomDrawerCloseOnClickAway
 
         #endregion
 
+        #region open drawer events/callbacks
+
+        public static readonly RoutedEvent DrawerOpenedEvent =
+            EventManager.RegisterRoutedEvent(
+                "DrawerOpened",
+                RoutingStrategy.Bubble,
+                typeof(DrawerOpenedEventHandler),
+                typeof(DrawerHost));
+
+        /// <summary>
+        /// Raised when a drawer is opened.
+        /// </summary>
+        public event DrawerOpenedEventHandler DrawerOpened
+        {
+            add { AddHandler(DrawerOpenedEvent, value); }
+            remove { RemoveHandler(DrawerOpenedEvent, value); }
+        }
+
+        /// <summary>
+        /// Attached property which can be used on the <see cref="Button"/> which instigated the <see cref="OpenDrawerCommand"/> to process the event.
+        /// </summary>
+        public static readonly DependencyProperty DrawerOpenedAttachedProperty = DependencyProperty.RegisterAttached(
+            "DrawerOpenedAttached", typeof(DrawerOpenedEventHandler), typeof(DrawerHost), new PropertyMetadata(default(DrawerOpenedEventHandler)));
+
+        public static void SetDrawerOpenedAttached(DependencyObject element, DrawerOpenedEventHandler value)
+        {
+            element.SetValue(DrawerOpenedAttachedProperty, value);
+        }
+
+        public static DrawerOpenedEventHandler GetDrawerOpenedAttached(DependencyObject element)
+        {
+            return (DrawerOpenedEventHandler)element.GetValue(DrawerOpenedAttachedProperty);
+        }
+
+        public static readonly DependencyProperty DrawerOpenedCallbackProperty = DependencyProperty.Register(
+            nameof(DrawerOpenedCallback), typeof(DrawerOpenedEventHandler), typeof(DrawerHost), new PropertyMetadata(default(DrawerOpenedEventHandler)));
+
+        /// <summary>
+        /// Callback fired when the <see cref="DrawerOpened"/> event is fired, allowing the event to be processed from a binding/view model.
+        /// </summary>
+        public DrawerOpenedEventHandler DrawerOpenedCallback
+        {
+            get { return (DrawerOpenedEventHandler)GetValue(DrawerOpenedCallbackProperty); }
+            set { SetValue(DrawerOpenedCallbackProperty, value); }
+        }
+
+        protected void OnDrawerOpened(DrawerOpenedEventArgs eventArgs)
+        {
+            RaiseEvent(eventArgs);
+        }
```

Comment:
I think for this issue we don't need the additonal callback. I think just the DrawerOpened routed event is fine.

Reply:
also for the closing callback?